### DISCUSSION TITLE
Dúvida sobre debug

### DISCUSSION THREAD

**Discussion by jpldaros** [balta-io/forum#454] [Tira Dúvidas] [answered]
Boa noite!

Estou utilizando a versão 8.0.100 do .NET.

A pasta .vscode não parece pra mim, não mostra a caixinha de pergunta na parte inferior da tela e quando executo com debug (ctrl+shift+D) não consigo criar essa pasta. 
Quando eu clico para criar, não aparecem opções .NET (como por exemplo a .NET Core, que aparece no seu vídeo), mesmo eu tendo instalado as extensões c# e .NET (quando parece a opção de instalar uma extensão e eu clico, já as tenho instaladas). Já reiniciei o VS Code diversas vezes e não adianta.

Poderia me ajudar, por favor?

Abraços

**Comment by andrebaltieri** [comment, marked as answer]
Boas noite, @jpldaros, como vai?

Recomendamos que utilize as mesmas versões do curso para maior compatibilidade.
Em relação a pasta .vscode, me parece que tem algum problema com o .NET na sua máquina.

Consegue mandar um print do VSCode?

**Reply by jpldaros** [reply]
Bom dia!

Certo, irei reinstalar com a mesma versão.

![imagem_2024-01-03_115558863](https://github.com/balta-io/forum/assets/132957131/734930d8-05da-43e7-b615-716d9c95e2ee)

Seria um print dessa tela de debug?

![imagem_2024-01-03_115751352](https://github.com/balta-io/forum/assets/132957131/398735b4-be86-457f-aaea-89feeb786344)

![image](https://github.com/balta-io/forum/assets/132957131/5ae0e801-1215-4806-88f9-dbeac20110c0)

![image](https://github.com/balta-io/forum/assets/132957131/a126da20-c5ea-499a-95ef-486dcc9893bd)

Mandei alguns prints a mais, por via das dúvidas.

**Reply by jpldaros** [reply]
Outro ponto, acredito que seja por conta da versão, que está me deixando muito confuso é em relação à escrita dos códigos. Segue print da aula "Escopo de programa" do módulo "Programação em C#", em comparação com o código pronto que aparece para mim, no VS Code.

![imagem_2024-01-03_121115702](https://github.com/balta-io/forum/assets/132957131/b9f993e8-6f16-4b23-8067-a7499f6292fa)

**Reply by andrebaltieri** [reply]
@jpldaros só usar o flag `--use-program-main` na hora de criar o projeto!
Por exemplo:

```
dotnet new console -o MeuProjeto --use-program-main
```

De resto acredito que nenhum código mudou, correto?

**Reply by jpldaros** [reply]
Perfeito! Fiz o teste e ficou igual.
De resto os códigos estão iguais sim.

Mas ainda continuo com o problema de debug.

Segue print:

![image](https://github.com/balta-io/forum/assets/132957131/e81c5ae3-fbaa-441a-a7f7-2f5bdbffdc43)

E quando eu clico nessa opção .NET 5+ and .NET Core, não acontece nada.

Devo reinstalar o .NET?

**Reply by andrebaltieri** [reply]
@jpldaros ele não cria o arquivo launchsettings.json na pasta properties?
![image](https://github.com/balta-io/forum/assets/965305/e13b321b-0640-4726-beae-f747e40e7d48)

Estas são as extensões que uso no VS Code
![image](https://github.com/balta-io/forum/assets/965305/62471875-2365-455f-aa71-4b1537fb5566)

**Reply by jpldaros** [reply]
Boa noite! 
Desculpe pela demora.

Não consegui achar essa pasta properties.

![image](https://github.com/balta-io/forum/assets/132957131/27527be3-4b7e-4949-8942-5a1cfb8df4e4)

Cacei tudo aqui, e só tenho essa pasta launch.json dentro do .vscode, não tenho nada de launchSettings.json

E essas extensões que você mandou eu utilizo também.

**Comment by andrebaltieri** [comment]
Bom dia, @jpldaros, como vai? 

Veja no vídeo (Gravei agora) como eu crio e debugo com launch.json
https://github.com/balta-io/forum/assets/965305/663e254f-2b8e-47f9-82bc-d3ba1223f7c5

**Reply by jpldaros** [reply]
Boa noite!

Certo. Essa pasta de properties eu não tenho. Devo ter feito algo errado ou deu algum problem no meio do caminho, vou reinstalar. Vou voltar às aulas iniciais para ver onde acho o erro também.

Muito obrigado pela paciência e atenção.

**Reply by jpldaros** [reply]
É, reinstalei tudo de novo e não funcionou.

![image](https://github.com/balta-io/forum/assets/132957131/735c3dcd-b9cc-4e44-a248-73bb5c0814d0)

![image](https://github.com/balta-io/forum/assets/132957131/397b3962-bc47-4711-9323-892b657bad08)

![image](https://github.com/balta-io/forum/assets/132957131/1fdf5b22-6fc3-45f2-be4d-cd88162e60cd)

![image](https://github.com/balta-io/forum/assets/132957131/31a29651-a761-48b4-bbc2-a1b469c2d1fa)

Realmente não sei o que posso fazer, procurei na documentação e também não consegui.

**Reply by andrebaltieri** [reply]
Quando você seleciona o "C#" ali na opção da última imagem, o que ocorre?
Você está com a extensão do C# instalada? E o DevKit?

**Reply by jpldaros** [reply]
Bom dia!

Não acontece nada, acho muito estranho. Eu clico e a lista suspensa some.

Sim, tô com ambas as extensões baixadas, C# e C# Dev Kit.

**Reply by andrebaltieri** [reply]
Eita, estranho mesmo... a última opção que vejo é remover o .NET, VS Code e reinstalar tudo novamente 🥲

**Reply by jenniferpessoa** [reply]
Oi, @jpldaros 
Veja se você precisa inserir manualmente as configurações no arquivo launch.json.

`"version": "0.2.0",
    "configurations": [
        {
            "name": ".NET Core Launch (console)",
            "type": "coreclr",
            "request": "launch",
            "preLaunchTask": "build",
            "program": "${workspaceFolder}/bin/Debug/NOMEDAPASTA/NOMEDOARQUIVO.dll",
            "args": [],
            "cwd": "${workspaceFolder}",
            "stopAtEntry": false,
            "console": "internalConsole"
        }
    ]`

![image](https://github.com/balta-io/forum/assets/83288893/e5e4835b-843a-4635-80ed-a44fb8825611)

**Reply by jenniferpessoa** [reply]
https://code.visualstudio.com/docs/editor/debugging
